pkg/virt-handler/cgroup/cgroup_test.go

```diff
@@ -0,0 +1,267 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021
+ *
+ */
+
+package cgroup
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/cilium/ebpf/asm"
+)
+
+type ProcCgroupData struct {
+	id         int
+	controller string
+	slice      string
+}
+
+var _ = Describe("Cgroup", func() {
+	var (
+		cgroupFS      string
+		procFS        string
+		procCgroupFmt string
+
+		major = int64(10)
+		minor = int64(200)
+	)
+
+	mockCgroupFS := func() (baseDir string) {
+		baseDir, err := ioutil.TempDir("", "cgroupfs-*")
+		Expect(err).ToNot(HaveOccurred())
+		return
+	}
+
+	mockProcFS := func() (baseDir string, pidCgroupFmt string) {
+		baseDir, err := ioutil.TempDir("", "procfs-*")
+		Expect(err).ToNot(HaveOccurred())
+		pidFmt := filepath.Join(baseDir, "%d")
+		err = os.MkdirAll(fmt.Sprintf(pidFmt, os.Getpid()), os.ModePerm)
+		Expect(err).ToNot(HaveOccurred())
+		pidCgroupFmt = filepath.Join(pidFmt, "cgroup")
+		return
+	}
+
+	prepareProcCgroupData := func(pidCgroupFmt string, data []ProcCgroupData) {
+		f, err := os.Create(fmt.Sprintf(pidCgroupFmt, os.Getpid()))
+		Expect(err).ToNot(HaveOccurred())
+		defer f.Close()
+		for _, d := range data {
+			_, err := fmt.Fprintf(f, "%d:%s:%s\n", d.id, d.controller, d.slice)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	}
+
+	BeforeEach(func() {
+		cgroupFS = mockCgroupFS()
```

Comment:
Do you think you can introduce structs which you can replace? I would really prefer side-effect tests where one does not have to replace package-level functions.

Reply:
hm.. not sure if I understood correctly. `cgroupFS` is a tests-local var defined above. It is then passed to the constructor of the respective struct. Currently there is no function replacement. Or maybe I missed something?

Reply:
Ah yes. Perfect. Never mind then :)